numba/unicode.py

```diff
@@ -1361,6 +1361,26 @@ def impl(a):
     return impl
 
 
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L11896-L11925    # noqa: E501
+@overload_method(types.UnicodeType, 'isspace')
+def unicode_isspace(data):
+    """Implements UnicodeType.isspace()"""
+
+    def impl(data):
+        length = len(data)
+        if length == 0:
+            return False
+
```

Comment:
CPython impl has a single char shortcut, perhaps implement that?

Reply:
As I see the shortcut exactly repeats body of the main loop. Do you think we still need to add the following code to the implementation?
```
if length == 1:
    code_point = _get_code_point(data, 0)
    if _PyUnicode_IsSpace(code_point):
        return True
    return False
```

Reply:
@stuartarchibald what do you think about above comment?

Reply:
It does reproduce the first iteration of the loop body. I'm wondering why CPython has this pattern, do you have any insight into this? I'm wondering if CPython is trying to avoid the control flow penalty or if the single character case is really common so it's just pulled out.

Reply:
I've had a bit of a think/realisation. See this comment: https://github.com/numba/numba/pull/4843#discussion_r346745393